plugins/modules/cloud/misc/proxmox_kvm.py

```diff
@@ -1085,20 +1078,15 @@ def main():
             if not vm:
                 module.exit_json(changed=False, msg="VM %s does not exist" % vmid)
 
-            if getattr(proxmox.nodes(vm[0]['node']), VZ_TYPE)(vmid).status.current.get()['status'] == 'running':
+            proxmox_node = proxmox.nodes(vm[0]['node'])
+            if getattr(proxmox_node, VZ_TYPE)(vmid).status.current.get()['status'] == 'running':
                 module.exit_json(changed=False, msg="VM %s is running. Stop it before deletion." % vmid)
-
-            taskid = getattr(proxmox.nodes(vm[0]['node']), VZ_TYPE).delete(vmid)
-            while timeout:
-                if (proxmox.nodes(vm[0]['node']).tasks(taskid).status.get()['status'] == 'stopped' and
-                        proxmox.nodes(vm[0]['node']).tasks(taskid).status.get()['exitstatus'] == 'OK'):
-                    module.exit_json(changed=True, msg="VM %s removed" % vmid)
-                timeout -= 1
-                if timeout == 0:
-                    module.fail_json(msg='Reached timeout while waiting for removing VM. Last line in task before timeout: %s'
-                                     % proxmox.nodes(vm[0]['node']).tasks(taskid).log.get()[:1])
-
-                time.sleep(1)
+            taskid = getattr(proxmox_node, VZ_TYPE).delete(vmid)
+            if not wait_for_task(module, proxmox, vm[0]['node'], taskid):
+                module.fail_json(msg='Reached timeout while waiting for removing VM. Last line in task before timeout: %s' %
+                                 proxmox_node.tasks(taskid).log.get()[:1])
```

Comment:
Are you sure that `proxmox_node` is still up to date? I don't know how the API works, but maybe it is better/necessary to recompute `proxmode_node` here to get up-to-date information.

Reply:
The `proxmox_node` variable represents the node in the Proxmox VE cluster rather than the destroyed VM which lets me think there is no need to recompute it.